app/assets/javascripts/omni-nav-v2.js

```diff
@@ -1,38 +1,109 @@
-var omni_nav_v2 = {
+var OmniNav2 = (function() {
 
-  initialize: function() {
-    this.$container = $('#omni-nav-v2');
-    this.$utility_nav = this.$container.find('.utility-nav');
-    this.$primary_nav = this.$container.find('#primary-nav');
 
+  var $container, $utility_nav, $primary_nav;
 
-    $('.utility-nav-trigger').on('click', omni_nav_v2.onUtilityNavClick);
-    this.$utility_nav.find('li.utility-has-dropdown').on('click', omni_nav_v2.onUtilityNavDropdownClick);
+  var initialize = function(container, utility_nav, primary_nav) {
+    $container = container;
+    $utility_nav = utility_nav;
+    $primary_nav = primary_nav;
+    $('.utility-nav-trigger').on('click', onUtilityNavClick);
+    $utility_nav.find('li.utility-has-dropdown').on('click', onUtilityNavDropdownClick);
+    $utility_nav.find('input:text').on('input', onSearchInput);
+    $primary_nav.find('input:text').on('input', onSearchInput); 
+  }
 
-  },
+  var onUtilityNavClick = function() {
+    $('.primary-nav-action').toggleClass("utility-open");
 
-  onUtilityNavClick : function() {
     if($(window).width() >= 768 ){
-      omni_nav_v2.$utility_nav.toggleClass('utility-nav-open');
-      $('html.omni-nav-v2').toggleClass('utility-nav-open');
-    }else {
-      omni_nav_v2.$primary_nav.toggleClass('search-open');
+      $utility_nav.toggleClass('utility-nav-open');
+      $('html.omni-nav-v2').toggleClass('utility-nav-open');  
+      $primary_nav.removeClass('search-open');    
+    } else {
```

Comment:
I actually prefer:

```
}
else {
```

But according to our guide, [cuddled elses](https://github.com/airbnb/javascript#blocks--cuddled-elses) are preferred. So feel free to call me on it when I fail to cuddle my elses.